File: src/main/java/me/nylestroke/nylemod/command/HomeCommand.java
package me.nylestroke.nylemod.command;

import com.mojang.brigadier.CommandDispatcher;
import me.nylestroke.nylemod.util.IEntityDataSaver;
import net.minecraft.server.command.CommandManager;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.text.LiteralText;
import net.minecraft.util.math.BlockPos;

public class HomeCommand {

    public static void register(CommandDispatcher<ServerCommandSource> dispatcher, boolean dedicated) {
        dispatcher.register(CommandManager.literal("sethome").executes((context -> {

            IEntityDataSaver player = (IEntityDataSaver) context.getSource().getPlayer();
            BlockPos playerPos = context.getSource().getPlayer().getBlockPos();
            String pos = "(" + playerPos.getX() + ", " + playerPos.getY() + ", " + playerPos.getZ() + ")";

            player.getPersistentData().putIntArray("homePos",
                    new int[]{ playerPos.getX(), playerPos.getY(), playerPos.getZ() });

            context.getSource().sendFeedback(new LiteralText("Set home at " + pos), true);
            return 1;

        })));

        dispatcher.register(CommandManager.literal("home").executes(context -> {
            IEntityDataSaver player = (IEntityDataSaver)context.getSource().getPlayer();

            // not 0 means in contains SOMETHING
            int[] homePos = player.getPersistentData().getIntArray("homePos");

            if (homePos.length != 0) {
                int[] playerPos = player.getPersistentData().getIntArray("homePos");
                context.getSource().getPlayer().requestTeleport(playerPos[0], playerPos[1], playerPos[2]);

                context.getSource().sendFeedback(new LiteralText("Player returned Home!"), true);
                return 1;
            } else {
                context.getSource().sendFeedback(new LiteralText("No Home Position has been Set!"), true);
                return -1;
            }

        }));
    }

}

File: src/main/kotlin/me/nylestroke/nylemod/command/HomeCommand.kt
package me.nylestroke.nylemod.command

import com.mojang.brigadier.Command
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import me.nylestroke.nylemod.util.IEntityDataSaver
import net.minecraft.command.CommandRegistryAccess
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.CommandManager.RegistrationEnvironment
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.text.Text
import net.minecraft.util.math.BlockPos

object HomeCommand {
    fun register(serverCommandSourceCommandDispatcher: CommandDispatcher<ServerCommandSource>,
                 commandRegistryAccess: CommandRegistryAccess,
                 registrationEnvironment: RegistrationEnvironment) {
        serverCommandSourceCommandDispatcher.register(CommandManager.literal("sethome").executes(Command<ServerCommandSource?>
        { context: CommandContext<ServerCommandSource?> ->
            val player: IEntityDataSaver = context.source?.player as IEntityDataSaver
            val playerPos: BlockPos? = context.source!!.player?.blockPos
            val pos = "(" + playerPos!!.x + ", " + playerPos!!.y + ", " + playerPos.z + ")"
            player.getPersistentData().putIntArray("homePos", intArrayOf(playerPos.x, playerPos.y, playerPos.z))
            context.source?.sendFeedback(Text.literal("Set home at $pos"), true)
            1
        }))

        serverCommandSourceCommandDispatcher.register(CommandManager.literal("home").executes(Command<ServerCommandSource?>
        executes@{ context: CommandContext<ServerCommandSource?> ->
            val player: IEntityDataSaver = context.source?.player as IEntityDataSaver

            // not 0 means in contains SOMETHING
            val homePos: IntArray = player.getPersistentData().getIntArray("homePos")
            if (homePos.isNotEmpty()) {
                val playerPos: IntArray = player.getPersistentData().getIntArray("homePos")
                context.source!!.player?.requestTeleport(playerPos[0].toDouble(), playerPos[1].toDouble(), playerPos[2].toDouble())
                context.source!!.sendFeedback(Text.literal("Player returned Home!"), true)
                return@executes 1
            } else {
                context.source!!.sendFeedback(Text.literal("No Home Position has been Set!"), true)
                return@executes -1
            }
        }))
    }
}